gms/gossiper.cc

```diff
@@ -492,36 +516,37 @@ rpc::no_wait_type gossiper::background_msg(sstring type, noncopyable_function<fu
 }
 
 void gossiper::init_messaging_service_handler() {
-    _messaging.register_gossip_digest_syn([this] (const rpc::client_info& cinfo, gossip_digest_syn syn_msg) {
+    _messaging.register_gossip_digest_syn([this] (const rpc::client_info& cinfo, gossip_digest_syn syn_msg, rpc::optional<instance_id_t> instance_id) {
```

Comment:
We should put instance_id into cinfo, so that it's available for all rpc calls. It will also have no per-call overhead on wire this way.

Reply:
It depends on if we really need this info for the other rpc calls. For the gossip usage, the per-call overhead is nothing. But if we need this for the read / write verbs, yes, cinfo is better.

Reply:
Didn't review yet, but we could have a separate RPC call when the connection is established to verify the node is permitted to connect. We don't need it on every call (IIUC).

Reply:
We could reuse CLIENT_ID for that.